pennylane/devices/qubit/apply_operation.py

```diff
@@ -79,29 +83,45 @@ def apply_operation_einsum(op: qml.operation.Operator, state):
     for old, new in zip(affected_indices, new_indices):
         new_state_indices = new_state_indices.replace(old, new)
 
-    einsum_indices = f"{new_indices}{affected_indices},{state_indices}->{new_state_indices}"
+    einsum_indices = (
+        f"...{new_indices}{affected_indices},...{state_indices}->...{new_state_indices}"
+    )
 
-    reshaped_mat = math.reshape(mat, [2] * (num_indices * 2))
+    new_mat_shape = [2] * (num_indices * 2)
+    if len(mat.shape) != 2:
+        # Add broadcasting dimension to shape
+        new_mat_shape.insert(0, mat.shape[0])
+    reshaped_mat = math.reshape(mat, new_mat_shape)
 
     return math.einsum(einsum_indices, reshaped_mat, state)
 
 
-def apply_operation_tensordot(op: qml.operation.Operator, state):
+def apply_operation_tensordot(op: qml.operation.Operator, state, batch_dim=0):
     """Apply ``Operator`` to ``state`` using ``math.tensordot``. This is more efficent at higher qubit
     numbers.
 
     Args:
         op (Operator): Operator to apply to the quantum state
         state (array[complex]): Input quantum state
+        batch_dim (int): Whether or not the state has a batch dimension. ``batch_dim`` should
+        be 0 if the state is unbatched, or 1 if the state is batched.
 
     Returns:
         array[complex]: output_state
     """
     mat = op.matrix()
-    total_indices = len(state.shape)
+    total_indices = len(state.shape) - batch_dim
     num_indices = len(op.wires)
-    reshaped_mat = math.reshape(mat, [2] * (num_indices * 2))
-    axes = (tuple(range(num_indices, 2 * num_indices)), tuple(op.wires))
+
+    new_mat_shape = [2] * (num_indices * 2)
+    if is_mat_batched := op.batch_size is not None:
```

Comment:
Think I just made the suggestion above. I think it would be worth while just using `op.batch_size is not None` everywhere instead of adding another variable to keep track of.